modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/networkapi/NetworkApiService.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.rosetta.domain.networkapi
+
+import cats.data.EitherT
+import cats.effect.Async
+
+import org.tessellation.BuildInfo
+import org.tessellation.rosetta.domain.NetworkIdentifier
+import org.tessellation.rosetta.domain.error.{LatestSnapshotNotFound, NetworkApiError}
+import org.tessellation.rosetta.domain.networkapi.model.options._
+import org.tessellation.rosetta.domain.networkapi.model.status._
+import org.tessellation.schema.{GlobalSnapshot, SnapshotOrdinal}
+import org.tessellation.security.Hashed
+import org.tessellation.security.hex.Hex
+
+trait NetworkApiService[F[_]] {
+  def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]]
+  def options: EitherT[F, NetworkApiError, NetworkApiOptions]
+  def status: EitherT[F, NetworkApiError, NetworkStatusResponse]
+}
+
+object NetworkApiService {
+  private val rosettaVersion = "1.4.14" // TODO get from a config?
+  private val version = Version(rosettaVersion = rosettaVersion, nodeVersion = BuildInfo.version)
+
+  def make[F[_]: Async](
+    getLastSnapshot: () => F[Option[Hashed[GlobalSnapshot]]],
+    getGenesisSnapshot: () => F[GlobalSnapshot],
+    getTimestampForSnapshot: SnapshotOrdinal => F[Long]
```

Comment:
do you need timestamp for each snapshot ordinal or just for the latest?

Reply:
The docs seem to specify the timestamp only for the current block.
https://www.rosetta-api.org/docs/models/NetworkStatusResponse.html

Reply:
got it. so I think that you should specify a function for the current block only. the way it is right now suggests that we should calculate timestamp for any arbitrary snapshot

Reply:
@marcinwadon  `s/getTimestampForSnapshot/getLastSnapshotTimestamp`  ?

Reply:
something like that (and then without argument?) 🤔

Reply:
also, you can create a newtype for timestamp

Reply:
What's the benefit of creating a newtype? are we going to constraint the values in some way?

Reply:
newtype is not for exactly for putting constraints. it's more about eliminating human error - if you see the type `Long` it can be anything (and you can use it anywhere). if you see the type `SnapshotOrdinal` (even if it's `Long` underlaying), you cannot misplace or misunderstand the type. hence, making it a `Timestamp` will differentiate it from (for example) snapshot ordinal, which is also of type `Long` but already wrapped into newtype.
also, newtypes are zero-cost runtime wrappers, so it's fine to use them in many places.

Reply:
@marcinwadon Back to the `getLastSnapshotTimestamp` ...

If we're going to call it that and remove the argument then it doesn't make sense to me that it be a separate function from obtaining last snapshot itself since there is a dependency between the two.

So wouldn't merging the two make sense? E.g. something like
```
getLastSnapshotInfo: () => F[Option[(Hashed[GlobalSnapshot], SnapshotTimestamp)]]
```

Reply:
yup, it makes perfect sense. although, `GlobalSnapshot` should be `GlobalIncrementalSnapshot` (but you need to base the change to the `feature/incremental-snapshots` branch)

Reply:
I noticed that. Thanks!